tools/crm_resource.c

```diff
@@ -965,7 +965,70 @@ main(int argc, char **argv)
         rc = pcmk_ok;
 
     } else if (rsc_cmd == 'e') {
+        xmlXPathObject *xpathObj = NULL;
+        xmlNode *cib_constraints = NULL;
+        crm_time_t *now = crm_time_new(NULL);
+        int i;
+
+        cib_constraints = get_object_root(XML_CIB_TAG_CONSTRAINTS,
+                                          data_set->input);
+        xpathObj = xpath_search(cib_constraints, "//" XML_CONS_TAG_RSC_LOCATION);
+
+        for (i = 0; i < numXpathResults(xpathObj); i++) {
+            xmlNode *constraint_node = getXpathResult(xpathObj, i);
+            xmlNode *date_expr_node = NULL;
+            crm_time_t *end = NULL;
+            char *xpath_string = NULL;
+
+            if (!crm_starts_with(ID(constraint_node), "cli-ban-") ||
+                !crm_starts_with(ID(constraint_node), "cli-prefer-")) {
```

Comment:
&&

Reply:
Hah, yes, I realized this while driving home and was hoping I could comment before anyone else noticed.  I swear I've programmed before.